File: ProjetAttractionAngular/src/app/model/construction.ts
export class Construction {

  constructor(
	  protected _nom : string,
	  protected _taille: number,
	  protected _prixAcquisition: number,
	  protected _prixFonctionnement: number,
	  protected _nbAmelioration: number,
	  protected _tauxIncident: number,
	  protected _affluence: number,
	  protected _id?: number
  ){}



	public get nom() : string {
		return this._nom;
	}


	public set nom(nom: string) {
		this._nom = nom;
	}


	public get prixAcquisition() : number {
		return this._prixAcquisition;
	}


	public set prixAcquisition(prixAcquisition: number) {
		this._prixAcquisition = prixAcquisition;
	}


	public get prixFonctionnement(): number {
		return this._prixFonctionnement;
	}


	public set prixFonctionnement(prixFonctionnement: number) {
		this._prixFonctionnement = prixFonctionnement;
	}



	public get nbAmelioration() : number{
		return this._nbAmelioration;
	}


	public set nbAmelioration(nbAmelioration: number) {
		this._nbAmelioration = nbAmelioration;
	}


	public get tauxIncident() : number{
		return this._tauxIncident;
	}


	public set tauxIncident(tauxIncident: number) {
		this._tauxIncident = tauxIncident;
	}


	public get taille(): number {
		return this._taille;
	}


	public set taille(taille: number) {
		this._taille = taille;
	}


	public get affluence() : number{
		return this._affluence;
	}


	public set affluence(affluence: number) {
		this._affluence = affluence;
	}


	public get id() : number | undefined {
    return this._id;
	}


	public set id(id : number|undefined) {
		this._id = id;
	}
}

File: ParcAttractionBoot/src/main/java/ParcAttractionBoot/model/Construction.java
package ParcAttractionBoot.model;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public abstract class Construction extends Element{

	@JsonView(JsonViews.Common.class)
	protected String nom;
	@JsonView(JsonViews.Common.class)
	protected double taille;
	@Column(name="prix_acquisition")
	@JsonView(JsonViews.Common.class)
	protected double prixAcquisition;
	@Column(name="prix_fonctionnement")
	@JsonView(JsonViews.Common.class)
	protected double prixFonctionnement;
	@Column(name="nb_amelioration") 
	@JsonView(JsonViews.Common.class)
	protected int nbAmelioration;
	@Column(name="taux_incident")
	@JsonView(JsonViews.Common.class)
	protected double tauxIncident;
	@Column(name="affluence_max")
	@JsonView(JsonViews.Common.class)
	protected int affluence;
	
	
	public Construction(Integer id, String nom, double prixAcquisition, double prixFonctionnement,
			int nbAmelioration, double tauxIncident, double taille, int affluence) {
		super(id);
		this.nom = nom;
		this.prixAcquisition = prixAcquisition;
		this.prixFonctionnement = prixFonctionnement;
		this.nbAmelioration = nbAmelioration;
		this.tauxIncident = tauxIncident;
		this.taille = taille;
		this.affluence = affluence;
	}
	
	
	public Construction(String nom, double prixAcquisition, double prixFonctionnement,
			int nbAmelioration, double tauxIncident, double taille, int affluence) {
		super();
		this.nom = nom;
		this.prixAcquisition = prixAcquisition;
		this.prixFonctionnement = prixFonctionnement;
		this.nbAmelioration = nbAmelioration;
		this.tauxIncident = tauxIncident;
		this.taille = taille;
		this.affluence = affluence;
	}
	
	public Construction() {}


	public String getNom() {
		return nom;
	}


	public void setNom(String nom) {
		this.nom = nom;
	}


	public double getPrixAcquisition() {
		return prixAcquisition;
	}


	public void setPrixAcquisition(double prixAcquisition) {
		this.prixAcquisition = prixAcquisition;
	}


	public double getPrixFonctionnement() {
		return prixFonctionnement;
	}


	public void setPrixFonctionnement(double prixFonctionnement) {
		this.prixFonctionnement = prixFonctionnement;
	}



	public int getNbAmelioration() {
		return nbAmelioration;
	}


	public void setNbAmelioration(int nbAmelioration) {
		this.nbAmelioration = nbAmelioration;
	}


	public double getTauxIncident() {
		return tauxIncident;
	}


	public void setTauxIncident(double tauxIncident) {
		this.tauxIncident = tauxIncident;
	}


	public double getTaille() {
		return taille;
	}


	public void setTaille(double taille) {
		this.taille = taille;
	}


	public int getAffluence() {
		return affluence;
	}


	public void setAffluence(int affluence) {
		this.affluence = affluence;
	}

	@Override
	public String toString() {
		return "Construction [id=" + id + ", nom=" + nom + ", prixAcquisition=" + prixAcquisition
				+ ", prixFonctionnement=" + prixFonctionnement
				+ ", nbAmelioration=" + nbAmelioration + ", tauxIncident=" + tauxIncident + ", taille=" + taille
				+ ", affluence=" + affluence + "]";
	}
	
	
	

}
